alcs-frontend/src/app/features/admin/holiday/holiday.component.ts

```diff
@@ -0,0 +1,101 @@
+import { AfterViewInit, Component, OnDestroy } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { PageEvent } from '@angular/material/paginator';
+import { BehaviorSubject, Subject, takeUntil } from 'rxjs';
+import { HolidayDto } from '../../../services/stat-holiday/holiday.dto';
+import { HolidayService } from '../../../services/stat-holiday/holiday.service';
+import { ConfirmationDialogService } from '../../../shared/confirmation-dialog/confirmation-dialog.service';
+import { HolidayDialogComponent } from './holiday-dialog/holiday-dialog.component';
+
+@Component({
+  selector: 'app-holiday',
+  templateUrl: './holiday.component.html',
+  styleUrls: ['./holiday.component.scss'],
+})
+export class HolidayComponent implements OnDestroy, AfterViewInit {
+  destroy = new Subject<void>();
+
+  public $statHolidays = new BehaviorSubject<HolidayDto[]>([]);
+  pageIndex = 0;
+  itemsPerPage = 20;
+  search?: number = undefined;
+  holidays: HolidayDto[] = [];
+  total: number = 0;
+  displayedColumns: string[] = ['name', 'day', 'actions'];
+
+  constructor(
+    private holidayService: HolidayService,
+    public dialog: MatDialog,
+    private confirmationDialogService: ConfirmationDialogService
+  ) {}
+
+  ngAfterViewInit(): void {
```

Comment:
Is there a reason why you are using AfterViewInit instead of OnInit?

Reply:
was a recommended way from angular to make sure that mat-table is initialized